app/utils/walletManagement.js

```diff
@@ -242,51 +241,107 @@ export function convertPeerValueToNumberValue(val) {
   return Number(val.replace(/[a-zA-Z]/gim, '').trim());
 }
 
-export async function getGlobalBoostStatistics(eosService) {
-  const limit = 100;
+export async function getUserBoostStatistics(
+  ethereumService,
+  user,
+  currentPeriod,
+) {
+  const averageStakeCurrent =
+    (await ethereumService.getTokenDataWithArgs(GET_AVERAGE_STAKE, [
+      currentPeriod,
+    ])) /
+    10 ** 18;
+  const averageStakeNext =
+    (await ethereumService.getTokenDataWithArgs(GET_AVERAGE_STAKE, [
+      currentPeriod + 1,
+    ])) /
+    10 ** 18;
+
+  const availableBalance =
+    (await ethereumService.getTokenDataWithArgs(GET_AVAILABLE_BALANCE, [
+      user,
+    ])) /
+    10 ** 18;
 
-  const { rows } = await eosService.getTableRows(
-    BOOST_STATISTICS_TABLE,
-    BOOST_STATISTICS_SCOPE,
-    undefined,
-    limit,
-    undefined,
-    undefined,
-    undefined,
-    process.env.EOS_TOKEN_CONTRACT_ACCOUNT,
+  const userStakeCurrent =
+    (await ethereumService.getTokenDataWithArgs(GET_USER_STAKE, [
+      user,
+      currentPeriod,
+    ])) /
+    10 ** 18;
+  const userStakeNext =
+    (await ethereumService.getTokenDataWithArgs(GET_USER_STAKE, [
+      user,
+      currentPeriod + 1,
+    ])) /
+    10 ** 18;
+
+  const userBoostCurrent =
+    (await ethereumService.getTokenDataWithArgs(GET_BOOST, [
+      user,
+      currentPeriod,
+    ])) / 1000;
+  const userBoostNext =
+    (await ethereumService.getTokenDataWithArgs(GET_BOOST, [
+      user,
+      currentPeriod + 1,
+    ])) / 1000;
+
+  const totalPeriodStake = await ethereumService.getTokenDataWithArgs(
+    GET_STAKE,
+    [currentPeriod + 1],
   );
 
-  return rows;
+  return {
+    averageStakeCurrent,
+    averageStakeNext,
+    availableBalance,
+    userStakeCurrent,
+    userStakeNext,
+    userBoostCurrent,
+    userBoostNext,
+    totalPeriodStake,
+  };
 }
 
-export async function getUserBoostStatistics(eosService, user) {
-  const limit = 100;
-
-  const { rows } = await eosService.getTableRows(
-    USER_BOOST_TABLE,
-    user,
-    undefined,
-    limit,
-    undefined,
-    undefined,
-    undefined,
-    process.env.EOS_TOKEN_CONTRACT_ACCOUNT,
+export async function getUserBoost(ethereumService, user, period) {
+  return Number(
+    (await ethereumService.getTokenDataWithArgs(GET_BOOST, [user, period])) /
+      1000,
   );
+}
 
-  return rows;
+export async function addBoost(ethereumService, user, tokens) {
+  await ethereumService.setStake(user, tokens);
 }
 
-export async function addBoost(eosService, user, tokens) {
-  await eosService.sendTransaction(
-    user,
-    ADD_BOOST_METHOD,
-    {
-      user,
-      tokens,
-    },
-    process.env.EOS_TOKEN_CONTRACT_ACCOUNT,
-    true,
-  );
+export function calculateNewBoost(userBoostStat, newStake) {
+  let predictedBoost;
+  let newAverageStake;
+  if (userBoostStat) {
```

Comment:
do a check at the beginning, then you will get rid of one nested if

```
if (!userBoostStat) {
  return [0, null]
}
```

Reply:
Done